src/DataViews/LiveFunctionsDataView.cpp

```diff
@@ -493,10 +493,21 @@ void LiveFunctionsDataView::OnDataChanged() {
       instrumented_functions = app_->GetCaptureData().instrumented_functions();
   for (const auto& [function_id, instrumented_function] : instrumented_functions) {
     const ModuleManager* module_manager = app_->GetModuleManager();
-    const FunctionInfo* function_info_from_capture_data =
-        orbit_client_data::FindFunctionByModulePathBuildIdAndOffset(
-            *module_manager, instrumented_function.file_path(),
-            instrumented_function.file_build_id(), instrumented_function.file_offset());
+    const FunctionInfo* function_info_from_capture_data;
+
+    // InstrumentedFunction::function_virtual_address() was added in 1.82: if this is not available,
+    // we need to keep using file_offset() to preserve compatibility with older captures.
+    if (instrumented_function.function_virtual_address() == 0) {
```

Comment:
Interesting. I'll have to think about the details a bit more. Do you have some more?
I guess we could go through the instrumented functions at the end of the capture and fill the information using the modules? Where would this happen? `OrbitApp::OnCaptureComplete` or `OrbitApp::OnCaptureFinished`?

I have to say I am a bit worried that this is going to be messier than the three `if`s added by this change but I guess it might be worth understanding this possibility more.

Reply:
So as the instrumented functions are in CaptureStarted. I would do that adjusting in OnCaptureStarted. I agree that this might become messy. My hope would be that it is actually pretty simple.

Reply:
I thought about it but I think that the problem is that to compute addresses from offsets we need the modules, and those come through `ModulesSnapshot` and `ModuleUpdateEvent`, which can be at any point during the capture and always after `CaptureStarted`.

Reply:
Makes sense. Thanks for the consideration.

Reply:
But maybe it can be done at the end of the capture. I'll look into it a bit more and come back to you.

Reply:
Good news! This can be done in `OrbitApp::OnCaptureComplete` like other post-processing we do, and it allows for further cleanups. I will merge this change as is as it works as is and create a separate one as I want a dedicated review and commit for that.

Reply:
https://github.com/google/orbit/pull/3947